dashboard/app/controllers/javabuilder_sessions_controller.rb

```diff
@@ -0,0 +1,58 @@
+require 'jwt'
+require 'securerandom' unless defined?(SecureRandom)
+require 'cdo/firehose'
+
+class JavabuilderSessionsController < ApplicationController
+  authorize_resource class: false
+
+  PRIVATE_KEY = CDO.javabuilder_private_key
+  PASSWORD = CDO.javabuilder_key_password
+
+  # GET /javabuilder/access_token
+  def get_access_token
+    channel_id = params[:channelId]
+    project_version = params[:projectVersion]
+    if !channel_id || !project_version
+      return render status: :bad_request, json: {}
+    end
+
+    begin
+      storage_id, storage_app_id = storage_decrypt_channel_id(channel_id)
+    rescue ArgumentError, OpenSSL::Cipher::CipherError
+      return render status: :bad_request, json: {}
+    end
+
+    issued_at_time = Time.now.to_i
+    # expire token in 15 minutes
+    expiration_time = (Time.now + 15.minutes).to_i
+    session_id = SecureRandom.hex(18)
+    payload = {
+      iat: issued_at_time,
```

Comment:
Do JWT tokens have a length limit? If not, might be useful for people reading the JavaBuilder code to read non-abbreviated attributes in the token (`issued_at_time` instead of `iat`).

Reply:
oops. I see these are "reserved claim names". Nevermind!